identity-iota/src/credential/validation_options.rs

```diff
@@ -0,0 +1,111 @@
+/// Options to declare validation criteria in
+/// [CredentialValidator::full_validation](super::CredentialValidator::full_validation()).
+#[non_exhaustive]
+pub struct CredentialValidationOptions {
+  /// Declares that the credential is **not** considered valid if it expires before this
+  /// [Timestamp].
+  pub earliest_expiry_date: Option<Timestamp>,
+  /// Declares that the credential is **not** considered valid if it was issued later than this
+  /// [Timestamp].
+  pub latest_issuance_date: Option<Timestamp>,
+  /// Declare that the credential's signature must be verified according to these
+  /// [VerifierOptions].
+  pub verifier_options: VerifierOptions,
+}
+
+impl CredentialValidationOptions {
+  /// Constructor that sets all options to their defaults.
+  pub fn new() -> Self {
+    Self::default()
+  }
+
+  /// Declare that a credential may expire no later than the given `timestamp`.
+  pub fn earliest_expiry_date(mut self, timestamp: Timestamp) -> Self {
+    self.earliest_expiry_date = Some(timestamp);
+    self
+  }
+  /// Declare that a credential may expire no later than the given `timestamp`.
+  pub fn latest_issuance_date(mut self, timestamp: Timestamp) -> Self {
+    self.latest_issuance_date = Some(timestamp);
+    self
+  }
+
+  /// Declare that the signature of a credential is to be verified according to the given
+  /// `options`.
+  pub fn verifier_options(mut self, options: VerifierOptions) -> Self {
+    self.verifier_options = options;
+    self
+  }
+}
+
+#[derive(Debug)]
+#[non_exhaustive]
+/// Options to declare validation criteria for
+/// [PresentationValidator::full_validation](super::PresentationValidator::full_validation()).
+pub struct PresentationValidationOptions {
+  /// Declares that the credentials of the presentation must all be
+  /// validated according to these options.
+  pub shared_validation_options: CredentialValidationOptions,
+  /// Declares that the presentation's signature is to be verified according to these
+  /// [VerifierOptions].
+  pub presentation_verifier_options: VerifierOptions,
+  pub(super) allow_non_transferable_violations: bool, // private as we may change the representation
+  pub(super) holder_must_be_subject: bool,            /* private as we may change the representation
+                                                       * note that holder_must_be_subject = true +
+                                                       * allow_non_transferable_violations = true can lead to
+                                                       * confusion
+                                                       * it would be better to introduce an enum
+                                                       * HolderSubjcetRelationShip {AlwaysSubject,
+                                                       * SubjectOnNonTransferable, DoNotValidate}
+                                                       * but it is not clear how we can expose that to javascript. */
```

Comment:
I would prefer that enum over the double bool here, and wasm-bindgen should allow for that, just like with `KeyType`.